airflow-core/src/airflow/models/asset.py

```diff
@@ -859,3 +860,58 @@ def __repr__(self) -> str:
         ]:
             args.append(f"{attr}={getattr(self, attr)!r}")
         return f"{self.__class__.__name__}({', '.join(args)})"
+
+
+class AssetPartitionDagRun(Base):
+    """
+    Keep track of new runs of a dag run per partition key.
+
+    Think of AssetPartitionDagRun as a provisional dag run. This record is created
+    when there's an asset event that contributes to the creation of a dag run for
+    this dag_id / partition_key combo. It may need to wait for other events before
+    it's ready to be created though, and the scheduler will make this determination.
+
+    We can look up the AssetEvents that contribute to AssetPartitionDagRun entities
+    with the PartitionedAssetKeyLog mapping table.
+
+    Where dag_run_id is null, the dag run has not yet been created.
+    We should not allow more than one like this. But to guard against
+    an accident, we should always work on the latest one.
+    """
+
+    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
+    target_dag_id: Mapped[str | None] = mapped_column(StringID(), nullable=False)
+    target_dag_run_id: Mapped[int | None] = mapped_column(Integer(), nullable=True)
+    partition_key: Mapped[str | None] = mapped_column(StringID(), nullable=False)
+    created_at: Mapped[datetime] = mapped_column(UtcDateTime, default=timezone.utcnow, nullable=False)
+    updated_at: Mapped[datetime] = mapped_column(
+        UtcDateTime, default=timezone.utcnow, onupdate=timezone.utcnow, nullable=False
+    )
+
+    __tablename__ = "asset_partition_dag_run"
+
+
+class PartitionedAssetKeyLog(Base):
+    """
+    Mapping table between AssetPartitionDagRun and AssetEvent.
+
+    PartitionedAssetKeyLog tells us which events contributed to a particular
+    AssetPartitionDagRun record.
+    """
+
+    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
+    asset_id: Mapped[int] = mapped_column(Integer, nullable=False)
+    asset_event_id: Mapped[int] = mapped_column(Integer, nullable=False)
+    asset_partition_dag_run_id: Mapped[int] = mapped_column(Integer, nullable=False)
+    source_partition_key: Mapped[str | None] = mapped_column(StringID(), nullable=False)
+    target_dag_id: Mapped[str | None] = mapped_column(StringID(), nullable=False)
+    target_partition_key: Mapped[str | None] = mapped_column(StringID(), nullable=False)
+    created_at: Mapped[datetime] = mapped_column(UtcDateTime, default=timezone.utcnow, nullable=False)
+
+    __tablename__ = "partitioned_asset_key_log"
+
+    def __repr__(self):
+        args = []
+        for attr in [x.name for x in self.__mapper__.primary_key]:
+            args.append(f"{attr}={getattr(self, attr)!r}")
+        return f"{self.__class__.__name__}({', '.join(args)})"
```

Comment:
```suggestion
        args = [
            f"{attr}={getattr(self, attr)!r}"
            for attr in [x.name for x in self.__mapper__.primary_key]
        ]
        return f"{self.__class__.__name__}({', '.join(args)})"
```

Reply:
actually this is another one we have in the codebase 
```
    def __repr__(self):
        args = [f"{x.name}={getattr(self, x.name)!r}" for x in self.__mapper__.primary_key]
        return f"{self.__class__.__name__}({', '.join(args)})"
```